docs/developer/mesh.md

```diff
@@ -0,0 +1,270 @@
+# Mesh
+
+Here we describe the two mesh formarts that are used in RDycore. The two mesh
+formats that we use are:
+
+1. Exodus II
+2. DMPlex HDF5 v3.0.0
+
+RDycore has been tested for meshes that include triangular, quadrilateral, or
+both triangular and quadrilateral cells. The vertices of cells have coordinates
+in 3D.
+
+## Exodus II Mesh Format
+
+The Exodus II mesh format (`.exo`) uses 1-based indices.
+
+### Triangular elements
+
+```text
+     Triangular
+        cell
+         v3
+         / \
+        /   \
+      e5     e4
+     /   c1   \
+    /          \
+  v1 --- e3 --- v2
+```
+
+- The `elem_type` of the triangular cell, `c1`, is `TRI3`
+- It comprises of three vertices (i.e. `v1`, `v2`, `v3`)
+- It has five edges sets. The first two edge sets form a plane and the remaining
+  three edge sets are lines as follows:
+  - `e1`: Comprises of the plane formed by `v1`, `v2`, `v3` (not shown above)
+  - `e2`: Comprises of the plane formed by `v1`, `v3`, `v2` (not shown above)
+  - `e3`: Line from `v1` to `v2`
+  - `e4`: Line from `v2` to `v3`
+  - `e5`: Line from `v3` to `v1`
+
+### Quadrilateral elements
+
+```text
+    Quadrilateral
+        cell
+  v4 --- e5 --- v3
+  |             |
+  |             | 
+  e6     c1     e4
+  |             | 
+  |             | 
+  v1 --- e3 --- v2
+```
+
+- The `elem_type` of the quadrilateral cell, `c1`, is `SHELL4`
+- It comprises of four vertices (i.e. `v1`, `v2`, `v3`, `v4`)
+- It has six edges sets. The first two edge sets form a plane and the remaining
+  four edge sets are lines as follows:
+  - `e1`: Comprises of the plane formed by `v1`, `v2`, `v3`, `v4` (not shown above)
+  - `e2`: Comprises of the plane formed by `v1`, `v4`, `v3`, `v2` (not shown above)
+  - `e3`: Line from `v1` to `v2`
+  - `e4`: Line from `v2` to `v3`
+  - `e5`: Line from `v3` to `v4`
+  - `e6`: Line from `v4` to `v1`
+
+### Example mesh with mixed element types
+
+![image](figures/quad_tri_mesh_exodus_ii.png)
+
+Above is an example of mesh that comprises of:
+
+- 12 cells with 4 quadrilaterals and 8 triangles
+- 14 vertices
+- Five edge sidesets
+  1. Right boundary
+  2. Left boundary
+  3. Top boundary
+  4. Bottom boundary
+  5. River
+
+The `.exo` mesh file uses NetCDF. The cells corresponding to the two element types
+(i.e. quadrilaterals and triangles) must be saved as two different data fields
+in the `.exo` file (e.g `connect1` for quadrilateral and `connect2` for triangles.
+
+```text
+ # quadrilateral cells
+ connect1 =
+  1, 2, 6, 5,
+  3, 4, 8, 7,
+  5, 6, 10, 9,
+  7, 8, 12, 11 ;
+
+# triangular cells
+ connect2 =
+  2, 3, 13,
+  3, 7, 13,
+  7, 6, 13,
+  6, 2, 13,
+  6, 7, 14,
+  7, 11, 14,
+  11, 10, 14,
+  10, 6, 14 ;  
+```
+
+Each sideset is defined via two data fields:
```

Comment:
Maybe we should briefly explain sidesets and how they relate to cells (elements) and edges at the top of the NetCDF format section.